File: node_api/src/vocabulary/RDFS.js
"use strict";
exports.__esModule = true;
exports.RDFS = void 0;
var RDFS = /** @class */ (function () {
    function RDFS() {
    }
    var _a;
    _a = RDFS;
    RDFS.NAMESPACE = "http://www.w3.org/2000/01/rdf-schema#";
    RDFS.PREFIX = "rdfs";
    RDFS.LABEL = RDFS.NAMESPACE + "label";
    RDFS.COMMENT = RDFS.NAMESPACE + "comment";
    RDFS.SUBCLASS = RDFS.NAMESPACE + "subClassOf";
    RDFS.PREFIXED = new /** @class */ (function () {
        function class_1(superThis) {
            var _this = this;
            this.superThis = superThis;
            Object.keys(superThis).forEach(function (key) { return _this[key] = superThis[key].replace(superThis.NAMESPACE, superThis.PREFIX + ":"); });
            return this;
        }
        return class_1;
    }())(_a);
    return RDFS;
}());
exports.RDFS = RDFS;

File: api/src/main/java/org/endeavourhealth/imapi/vocabulary/RDFS.java
package org.endeavourhealth.imapi.vocabulary;

import org.endeavourhealth.imapi.model.tripletree.TTIriRef;

import static org.endeavourhealth.imapi.model.tripletree.TTIriRef.iri;

public enum RDFS implements Vocabulary {
    NAMESPACE("http://www.w3.org/2000/01/rdf-schema#"),
    PREFIX("rdfs"),
    LABEL(NAMESPACE.iri +"label"),
    COMMENT(NAMESPACE.iri +"comment"),
    SUBCLASSOF(NAMESPACE.iri +"subClassOf"),
    SUBPROPERTYOF(NAMESPACE.iri +"subPropertyOf"),
    DOMAIN(NAMESPACE.iri +"domain"),
    RANGE(NAMESPACE.iri +"range"),
    RESOURCE(NAMESPACE.iri +"Resource"),
    CLASS(NAMESPACE.iri +"Class"),
    DATATYPE(NAMESPACE.iri +"Datatype"),
    IS_DEFINED_BY(NAMESPACE.iri +"isDefinedBy");

    public final String iri;
    RDFS(String iri) {
        this.iri = iri;
    }

    @Override
    public TTIriRef asTTIriRef() {
        return iri(this.iri);
    }

    @Override
    public String getIri() {
        return iri;
    }

    public static boolean contains(String iri) {
        try {
            RDFS.valueOf(iri);
            return true;
        } catch (IllegalArgumentException e) {
            return false;
        }
    }
}
